assets/app/data/crowbar/services/upgrade-status.factory.spec.js

```diff
@@ -187,6 +187,63 @@ describe('Upgrade Status Factory', function () {
                     });
                 });
             });
+
+            describe('when got upgrade status from api unsuccessfully', function () {
+                describe('when timeout == 0', function () {
+                    beforeEach(function () {
+                        bard.mockService(upgradeFactory, {
+                            getStatus: $q.reject()
+                        });
+
+                        upgradeStatusFactory.waitForStepToEnd(
+                            testedStep, mockedSuccessCallback, mockedErrorCallback, pollingInterval
+                        );
+
+                        $rootScope.$digest();
+                    });
+
+                    it('should not call success callback', function () {
+                        expect(mockedSuccessCallback).not.toHaveBeenCalled();
+                    });
+
+                    it('should call error callback', function () {
+                        expect(mockedErrorCallback).toHaveBeenCalled();
+                    });
+
+                    it('should not schedule another check', function () {
+                        expect(mockedTimeout).not.toHaveBeenCalled();
+                    });
+                });
+
+                describe('when timeout > 0', function () {
+                    beforeEach(function () {
+                        bard.mockService(upgradeFactory, {
+                            getStatus: $q.reject()
+                        });
+
+                        upgradeStatusFactory.waitForStepToEnd(
+                            testedStep, mockedSuccessCallback, mockedErrorCallback, 1, 10
+                        );
+
+                        spyOn(upgradeStatusFactory, 'waitForStepToEnd').and.callThrough();
```

Comment:
Ah yes, good point! 

Its a leftover. I was trying to validate to see if it was being called, but Im not sure I was getting that it wasnt called. 

I migth need to call $scope.$digest() or something like that due to this function being called by setTimeout?

Im not sure how to proceed to check if that was called tbh.